### PR TITLE
Add partition number in GEM trigger primitives

### PR DESCRIPTION
**Pull request by dildick**
#### PR description:

The new GE2/1 geometry has 16 eta partitions (each strip split in 2), instead of 8 eta partitions. To prevent a miscommunication between GE2/1 TPs and the GEM-CSC local trigger, the EMTF, or the GEM packer/unpacker, I added a member `part_` in the `GEMPadDigi` and `GEMPadDigiCluster`. This was requested by EMTF folks. This way it should be relatively easy to check which geometry you are dealing with by calling `nPartitions()`. (Retry of #30998.) 16-partition  GE2/1 pads are currently rejected in the GEM-CSC trigger and GEM validation.

This PR is to be followed with an update of the EMTF to reject 16-partition GE2/1 pads temporarily until algorithms or validation modules are updated. 

See also #https://github.com/cms-sw/cmssw/pull/31264.
 
#### PR validation:

Tested with WF 23234.0

```
23234.0_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Thu Aug 27 12:06:19 2020-date Thu Aug 27 11:33:38 2020; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed
```
#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A 

@watson-ij @bsunanda @jshlee @jiafulow @eyigitba

### CODE REVIEW

**Comment by dildick** [comment]
I noticed I forgot to include a case in the cluster producer.

**Comment by kpedro88** [comment]
please test

**Comment by dildick** [comment]
This should fix it 
```
12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Fri Aug 28 10:41:56 2020-date Fri Aug 28 10:03:45 2020; exit: 0 0 0 0 0
1 1 1 1 1 tests passed, 0 0 0 0 0 failed
```

**Comment by jfernan2** [comment]
please test

**Comment by dildick** [comment]
There was a small typo. I ran one of the WFs again

```
23234.0_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Fri Aug 28 15:46:00 2020-date Fri Aug 28 15:08:01 2020; exit: 0 0 0 0
1 1 1 1 tests passed, 0 0 0 0 failed
```

**Comment by kpedro88** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by dildick** [comment]
@jfernan2 This should run now. I ran a few tests
```
11634.0_TTbar_14TeV+2021+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Sun Aug 30 11:32:50 2020-date Sun Aug 30 10:44:33 2020; exit: 0 0 0 0 0
12434.0_TTbar_14TeV+2023+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED Step4-PASSED  - time date Sun Aug 30 11:32:52 2020-date Sun Aug 30 10:44:34 2020; exit: 0 0 0 0 0
23234.0_TTbar_14TeV+2026D49+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Sun Aug 30 11:28:23 2020-date Sun Aug 30 10:44:35 2020; exit: 0 0 0 0
28234.0_TTbar_14TeV+2026D60+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Sun Aug 30 11:34:12 2020-date Sun Aug 30 10:44:35 2020; exit: 0 0 0 0
4 4 4 4 2 tests passed, 0 0 0 0 0 failed
``` 
But I need to clean up the index. Some commits are messed up now.

**Comment by dildick** [comment]
I cleaned up the branch and I'll rerun a few tests.

**Comment by jfernan2** [comment]
please test

**Comment by dildick** [comment]
@jfernan2 Nice! It runs! 👍 That's it for this pull request then.

**Comment by cvuosalo** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

File: DataFormats/GEMDigi/interface/GEMPadDigi.h [hunk 1]
```diff
@@ -33,12 +37,18 @@ class GEMPadDigi {
   int16_t bx() const { return bx_; }
   GEMSubDetId::Station station() const { return station_; }
 
+  // Newer GE2/1 geometries will have 16! eta partitions
+  // instead of the usual 8.
+  void setNPartitions(unsigned nPart) { part_ = nPart; }
```
Comment (kpedro88) [inline DataFormats/GEMDigi/interface/GEMPadDigi.h:42]: will there ever be a case where the `part_` value needs to be changed after the digi object is created?

Comment (dildick) [reply DataFormats/GEMDigi/interface/GEMPadDigi.h:42]: Very unlikely. I added the function for completeness.

Comment (kpedro88) [reply DataFormats/GEMDigi/interface/GEMPadDigi.h:42]: There aren't setters for the other attributes... it might be safer to remove it.

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
@dildick it looks like the latest commit added some unwanted files. Can you overwrite it so these don't enter the history?

**Comment by kpedro88** [comment]
please test

**Comment by dildick** [comment]
Hi, can this pull request be integrated? Thanks.

**Comment by silviodonato** [comment]
@kpedro88 do you have further comments after https://github.com/cms-sw/cmssw/pull/31268/commits/58f5ad00d4c5e1342640fecc75584e9f08b374fb ?

**Comment by kpedro88** [comment]
+upgrade

**Comment by cvuosalo** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by dildick** [comment]
@civanch Can you sign off, please? Thanks!

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
@silviodonato,@qliphy please merge, @cms-sw/l1-l2 never signs

**Comment by qliphy** [comment]
+1

**Comment by qliphy** [comment]
merge

**Pull request merged**